code/modules/clothing/shoes/miscellaneous.dm

```diff
@@ -69,28 +69,46 @@
 	SEND_SIGNAL(t_loc, COMSIG_TURF_MAKE_DRY, TURF_WET_WATER, TRUE, INFINITY)
 
 /obj/item/clothing/shoes/clown_shoes
-	desc = "The prankster's standard-issue clowning shoes. Damn, they're huge!"
+	desc = "The prankster's standard-issue clowning shoes. Damn, they're huge! Ctrl-click to toggle waddle dampeners."
 	name = "clown shoes"
 	icon_state = "clown"
 	item_state = "clown_shoes"
 	slowdown = SHOES_SLOWDOWN+1
 	item_color = "clown"
 	pocket_storage_component_path = /datum/component/storage/concrete/pockets/shoes/clown
+	var/datum/component/waddle
```

Comment:
Holding component references is a bad, use a weakref if at all

Reply:
I'll need some help with this cause I just copied how it was done for the penguin ears, I'm still learning how to deal with components

Reply:
@Cyberboss

Reply:
why is holding component references bad